lux/lib/lux/prisms/discord/message_prism.ex

```diff
@@ -0,0 +1,281 @@
+defmodule Lux.Prisms.Discord.MessagePrism do
+  @moduledoc """
+  A prism for sending messages to Discord channels.
+
+  ## Authentication
+
+  Requires a Discord bot token in the application configuration:
+  ```elixir
+  config :lux, :discord_token, "your-bot-token"
+  ```
+
+  ## Rate Limiting
+
+  Discord API has rate limits per route. This prism handles rate limiting by:
+  - Respecting the retry_after header
+  - Following the global rate limit
+  - Using proper request headers
+
+  ## Example
+
+      iex> Lux.Prisms.Discord.MessagePrism.run(%{
+      ...>   channel_id: "123456789",
+      ...>   content: "Hello from Lux!",
+      ...>   embeds: [
+      ...>     %{
+      ...>       title: "Example Embed",
+      ...>       description: "This is a test embed",
+      ...>       color: 0x7289DA
+      ...>     }
+      ...>   ]
+      ...> })
+      {:ok,
+       %{
+         status: "success",
+         message: %{
+           id: "987654321",
+           channel_id: "123456789",
+           content: "Hello from Lux!",
+           timestamp: "2024-03-08T12:34:56Z"
+         }
+       }}
+
+  ## Error Cases
+
+      # Invalid channel ID
+      iex> Lux.Prisms.Discord.MessagePrism.run(%{channel_id: "invalid"})
+      {:error, %{type: "validation_error", message: "Invalid channel ID format"}}
+
+      # Missing permissions
+      iex> Lux.Prisms.Discord.MessagePrism.run(%{channel_id: "123456789"})
+      {:error, %{type: "discord_api_error", code: 50001, message: "Missing Access"}}
+
+      # Rate limited
+      iex> Lux.Prisms.Discord.MessagePrism.run(%{channel_id: "123456789"})
+      {:error, %{type: "rate_limit", retry_after: 5000}}
+
+  The prism reads Discord authentication details from configuration:
+  - :discord_token - Discord bot token for authentication
+  """
+
+  use Lux.Prism,
+    name: "Discord Message Sender",
+    description: "Sends messages to Discord channels",
+    url: "https://discord.com/api/v10/channels/:channel_id/messages",
```

Comment:
It looks like you are setting a lot of parameters that would be present in Lenses here. `url`, `method`, etc are not attributes of Prisms.

Reply:
If you try to compile this code, you indeed get a warning about this 
![image](https://github.com/user-attachments/assets/f4e3611f-2d53-4870-94da-70c992ebb828)